sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -49,9 +49,12 @@ case class HadoopFsRelation(
   override def sqlContext: SQLContext = sparkSession.sqlContext
 
   val schema: StructType = {
-    val dataSchemaColumnNames = dataSchema.map(_.name.toLowerCase).toSet
-    StructType(dataSchema ++ partitionSchema.filterNot { column =>
-      dataSchemaColumnNames.contains(column.name.toLowerCase)
+    val equality = sparkSession.sessionState.conf.resolver
+    val overriddenDataSchema = dataSchema.map { dataField =>
```

Comment:
Why didn't you use `sparkSession.sessionState.conf.resolver`? Any reason I missed?
I just wrote this code with the same style with `DataSource#getOrInferFileFormatSchema` and is this a bad idea? Anyway, since the two patterns have the same output, both is okay to me.